lib/syskit/network_generation/logger.rb

```diff
@@ -165,17 +165,17 @@ def self.add_logging_to_network(engine, work_plan)
                     # Disconnect current log connections, we're going to
                     # reestablish the ones we want later on. We leave other
                     # connections as-is
-                    unless seen_loggers.include?(logger_task)
-                        dataflow = work_plan.task_relation_graph_for(Flows::DataFlow)
-                        deployment.each_executed_task do |t|
-                            if engine.deployed_tasks.include?(t)
-                                dataflow.remove_relation(t, logger_task)
-                            end
+                    dataflow = work_plan.task_relation_graph_for(Flows::DataFlow)
+                    deployment.each_executed_task do |t|
+                        if engine.deployed_tasks.include?(t)
+                            dataflow.remove_relation(t, logger_task)
                         end
                     end
 
                     next if required_logging_ports.empty?
 
+                    required_loggers << logger_task
```

Comment:
so to disable logger in one of the deployment instances (when the deployment is shared), we just need to make `deployment.log_port?` false for every port its executed tasks have?

Reply:
That is the nexus for the decision. There is however already a system for that, described here: https://www.rock-robotics.org/rock-and-syskit/log_management/runtime.html